paper.md

```diff
@@ -58,202 +58,195 @@ bibliography: paper.bib
-   - These submodules cover all concepts necessary to understand
-   the lesson content and complete the take-home assignments and
-   in-class exercises. In lectures where the explicit goal was
-   not to learn programming, but a topic such as "Statistical
-   Modelling", we used the coding submodules to explain the concept
-   while concurrently demonstrating how to use programming to
-   create and apply statistical models to data.
-3. *Interleaved Exercises*: Coding exercises or discussion points
-during the lecture to assess and confirm that students are
-following along.
-   - These exercises serve to both challenge the students and slowly
-   build confidence in the lesson material.
-   - If many students struggle to successfully complete a given
-   exercise, this is a way for instructors to identify which
-   concepts to go over more thoroughly or clear up any misunderstandings.
-4. *Summative Assignment*: A comprehensive assignment to test the
-competency of students on the lesson material.
-   - Assignments were designed to challenge the students to apply
-   the techniques and concepts from the lesson to solve new problems,
-   demonstrating their understanding of the material.
-   - Each assignment covers two lectures worth of material.
-
-Each of our lectures built on skills and concepts that would ultimately allow
-students to work on a final open-ended analysis of real open ecological data.
-We deliberately chose large and messy (e.g. missing values) datasets for use in the course,
-reflecting the types of data that are being increasingly generated across
-various disciplines. With this goal in mind, we designed lectures to provide the
-building blocks to clean, manipulate, visualize, and analyze any datasets
-students may come across, including those used for the final projects.
+Drawing on the instructors' previous experiences teaching introductory
+programming workshops, we designed our lessons to have the following components:
+
+1. *Lesson Outline*: A clearly defined outline of the lesson objectives
+including expected time spent on each objective. This gives students a clear 
+expectation on what they should learn and gain from the lesson. It also provides
+a structured template for instructors to prioritize content and gauge how much 
+time each objective should take.
+2. *Participatory Live-Coding*: Coding in real-time, with the students actively
+following along, form the primary focus of each lesson material. This hands-on
+approach to teaching is frequently used by teaching organizations such as 
+[Software Carpentry](https://software-carpentry.org/blog/2016/04/tips-tricks-live-coding.html))
+[@carpentry;@rubin_effectiveness_2013;@haaranen_programming_2017;@wilson_teaching_2018].
+While most lessons' goals were to learn programming by doing, some lessons were
+heavier on concepts (such as "Statistical Modelling" or "Differential
+Equations"), during which we still used the live-coding approach. This allows 
```

Comment:
Change to `This approach not only demonstrates the concepts in a step-by-step fashion but also helps students practice...`